skmob/measures/home_detection.py

```diff
@@ -0,0 +1,420 @@
+        raise ValueError("Invalid week_period argument")
+
+    # if the specified TrajDataFrame is empty, return (NaN, Nan)
+    if traj.empty:
+        return (nan, nan)
+
+    # create a new column containing the home location activity
+    traj[constants.ACTIVITY] = 0
+
+    # calculate the number of occurences for (latitude, longitude) and sort the resulting DataFrame in descending order
+    traj = traj[[constants.LATITUDE, constants.LONGITUDE, constants.ACTIVITY]].groupby([constants.LATITUDE, constants.LONGITUDE]).count().sort_values(by=constants.ACTIVITY, ascending=False).reset_index()
+ 
+    # if a radius argument was specified, apply the related radius_activity algorithm
+    if isinstance(radius, (int, float)):
+        if mode == 'sjoin':
+            lat, lng = _sjoin_activity(traj, radius)
+        elif mode == 'distance_matrix':
+            lat, lng = _radius_activity(traj, radius)
+        else:
+            raise ValueError("Invalid mode argument")
+    # if no radius was specified, select and return the first row (latitude, longitude) from the processed DataFrame       
+    elif(radius == None):
+        lat = traj.at[0, constants.LATITUDE]
+        lng = traj.at[0, constants.LONGITUDE]
+    else:
+        raise ValueError("Invalid radius argument")
+    return (lat, lng)
+    
+
+def _home_location_tc(traj, week_period, radius, mode, start_time, end_time):
+    # set the datetime as index, apply the time filters and perform the mostAmount algorithm on the new DataFrame
+    traj = traj.set_index(pd.DatetimeIndex(traj.datetime)).between_time(start_time, end_time).reset_index(drop=True)
+    return _home_location_ma(traj, week_period, radius, mode)
+
+
+def _calculate_dd_activity(traj):
+    # return a tuple, containing the number of distinct days and the total number
+    distinct_days = traj[constants.DATETIME].dt.date.unique()
+    return (len(distinct_days), len(traj))
+
+
+def _home_location_dd(traj):
+    traj = traj.groupby([constants.LATITUDE, constants.LONGITUDE]).apply(lambda x: _calculate_dd_activity(x)).reset_index().rename(columns={0: constants.ACTIVITY})
+    traj = traj.sort_values(by=constants.ACTIVITY, ascending=False).reset_index()
+
+    lat = traj.at[0, constants.LATITUDE]
+    lng = traj.at[0, constants.LONGITUDE]
+
+    return (lat, lng)
+
+
+def _home_location_in(traj, threshold):
+    # create and calculate a new 'time_gap' column, containing the difference, in minutes, between sequential rows in the DataFrame
+    traj[constants.TIME_GAP] = traj.sort_values(constants.DATETIME, ascending=True)[constants.DATETIME].diff(1).dt.total_seconds().div(60).shift(-1)
+    
+    # filter the rows that are exceeding the specified threshold on the new DataFrame and apply the mostAmount algorithm
+    traj = traj[traj[constants.TIME_GAP] > threshold]
+    return _home_location_ma(traj, None, None, None)
+
+
+
+
+def home_location_ma(traj, week_period=None, radius=None, mode='sjoin', show_progress=True):
```

Comment:
I suggest to call this most_amount(..) and then it will be called as
```
from skmob.measures import home_detection
home_detection.most_amount(..)
```